File: frontend/src/types/itemImage.ts
// interface for uploading item image 
export interface PostItemImage {
  listing_id: string;
  file: any;
  is_primary?: boolean;
}

export interface ItemImageResponse {
  id: string;
  listing_id: string;
  image_url: string;
  is_primary: boolean;
  created_at?: string;
}

File: backend/app/schemas/item_image.py
from __future__ import annotations

from uuid import UUID
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field, ConfigDict
from backend.app.models import item_image

# Base item image schema    
class Item_ImageBase(BaseModel):
    is_primary: bool = Field(...)


# Create item image
class Item_ImageCreate(Item_ImageBase):
    pass

# Read/Response item image
class Item_ImageOut(Item_ImageBase):
    model_config = ConfigDict(from_attributes=True)
    image_id: UUID
    item_listing_id: UUID
    file_url: str
    is_primary: bool
    created_at: datetime
    
